packages/legacy-client/src/client/interfaces/objectSet.ts

```diff
@@ -14,54 +14,60 @@
  * limitations under the License.
  */
 
-import type {
-  ListObjectsError,
-  OntologyObject,
-  Page,
-  Result,
-} from "../../ontology-runtime";
+import type { ObjectTypesFrom, OntologyDefinition } from "@osdk/api";
+import type { ListObjectsError, Page, Result } from "../../ontology-runtime";
+import type { OsdkLegacyObjectFrom } from "../OsdkObject";
 import type { ObjectTypeFilterFunction } from "./filters";
 import type { OrderByFunction } from "./ordering";
 import type { SearchAround } from "./searchAround";
 
-export type ObjectSet<O extends OntologyObject> =
+export type ObjectSet<
```

Comment:
Should this type be internal? Something I'm concerned about is breaking the current experience of `const x : ObjectSet<Todo> = ...`  we may need to generate a bit more code to preserve some of the user facing types if we go this approach for these interfaces

Reply:
Yeah that's an interesting point, I guess we're kind of talking past each other a bit with the intention of these types. You intended them to be user-facing but I assumed they were going to be more geared to the output of the conversion process to make sure that all lined up.

Perhaps I should revert this set of changes and either 1) accept less type inference from the output step before it gets assigned the generated types, or 2) add a second set of types targeted towards the output before it gets wrapped.

@ericanderson how much do you care about the 'raw' output having type inference across links to different object types?